test/test_tensorexpr.py

```diff
@@ -1578,5 +1578,69 @@ def foo(a, b, c):
             exp = traced(a, b, c)
             self.assertEqual(ref, exp)
 
+    def test_1d_strided_within_bounds(self):
```

Comment:
I think `test/cpp/tensorexpr/test_expr.cpp` is a more appropriate place to add these tests. NNC is implemented in C/C++, and currently its unit tests (files in `test/cpp/tensorexpr`) are mainly written in C++. We are using `test/test_tensorexpr.py` to test the integration of NNC to JIT for op fusions.